build_tools/buildkite/agent/hooks/environment-trusted-agent.sh

```diff
@@ -0,0 +1,9 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+export BUILDKITE_ANNOTATION_CONTEXT="$${BUILDKITE_STEP_ID}"
+export IREE_BUILDKITE_ACCESS_TOKEN="$(gcloud secrets versions access latest \
+          --secret=iree-buildkite-privileged)"
```

Comment:
Will this cause the access token being passed down to everywhere?

I guess for presubmit this isn't a big issue as attacker can always submit a pipeline to get the presubmit token. But for the submitted pipeline we need to make sure the access token doesn't flow to the wrong place. Wondering if it is a little bit risky.

Reply:
It means that jobs running on the trusted agent will always have access to these in the environment. They already have access via the service account they're running as (they can make this gcloud call), so this doesn't extend their access at all. Environment variables are not passed to triggered pipelines, so that shouldn't cause any issues. I believe the current setup actually presents more of a risk because it increases the likelihood of accidentally printing the value of the secret to the log. Demos with a fake secrete and my test pipeline leaking the secret [directly with echo](https://buildkite.com/iree/gcmn-test-pipeline/builds/838) or (more likely) [with `set -x`](https://buildkite.com/iree/gcmn-test-pipeline/builds/840).